tests/test_states.py

```diff
@@ -0,0 +1,65 @@
+
+# pylint: disable=protected-access, unused-argument
+
+
+# ------------------------------------------------------------------------------
+#
+def test_pilot_state_value():
+    from radical.pilot.states import _pilot_state_value
+    assert(_pilot_state_value('NEW')                    == 0)
+    assert(_pilot_state_value('PMGR_LAUNCHING_PENDING') == 1)
+    assert(_pilot_state_value('PMGR_LAUNCHING')         == 2)
+    assert(_pilot_state_value('PMGR_ACTIVE_PENDING')    == 3)
+    assert(_pilot_state_value('PMGR_ACTIVE')            == 4)
+    assert(_pilot_state_value('DONE')                   == 5)
+    assert(_pilot_state_value('FAILED')                 == 5)
+    assert(_pilot_state_value('CANCELED')               == 5)
+
+# ------------------------------------------------------------------------------
+
+
+# ------------------------------------------------------------------------------
+#
+def test_pilot_state_collapse():
+    from radical.pilot.states import _pilot_state_collapse,_pilot_state_values
+
+    for key in _pilot_state_values.iterkeys():
+        assert(_pilot_state_collapse([key]) == key)
+# ------------------------------------------------------------------------------
```

Comment:
Looking over this again, I don't think this test covers the right cases.  The method is supposed to select the most current state out of a list of states.  Assume a unit gets state updates out of order (as it happens) - what is our best guess in what the state the unit is in?

```
[NEW, UMGR_STAGING_INPUT, FAILED, UMGR_STAGING_INPUT_PENDING] -> FAILED
[NEW, EXECUTING, UMGR_STAGING_INPUT, UMGR_STAGING_INPUT_PENDING] -> EXECUTING
```
etc.

Reply:
Aha! Thank you! I will update the test with a few test cases then that have states out of order and provide the last one!

Reply:
Thank you, @iparask !

Reply:
Should this still be addressed or  will that go into a different PR?  If the latter, please open a ticket to keep track of it.

Reply:
Fixing it!